src/js/components/sharedComponents/TooltipWrapper.jsx

```diff
@@ -0,0 +1,173 @@
+        }
+        else if (this.props.wide) {
+            // is there at least 801px of space to the left/right for the tooltip?
+            tooltipWidth = (window.innerWidth - tooltipContainer.offsetLeft > 800)
+                ? 700
+                : window.innerWidth - tooltipContainer.offsetLeft - 100;
+        }
+        let offsetRight = window.innerWidth - tooltipContainer.offsetLeft - tooltipContainer.clientWidth - tooltipWidth - 30;
+        if (this.props.left) {
+            offsetRight = (window.innerWidth - tooltipContainer.offsetLeft) + tooltipContainer.clientWidth;
+        }
+        this.setState({
+            offsetTop,
+            offsetRight,
+            width: tooltipWidth
+        });
+    }
+
+    render() {
+        const showTooltip = (this.props.controlledProps.isControlled) ? this.props.controlledProps.isVisible : this.state.showTooltip;
+        let tooltip = null;
+        const style = {
+            top: this.state.offsetTop,
+            right: this.state.offsetRight,
+            width: this.state.width
+        };
+        if (showTooltip) {
+            tooltip = (
+                <div className="tooltip-spacer" style={style}>
+                    <div className="tooltip" id="tooltip" role="tooltip">
+                        <div className="tooltip__interior">
+                            <div
+                                className={`tooltip-pointer ${
+                                    this.props.left ? "right" : ""
+                                }`} />
+                            <div className="tooltip__content">
+                                <div className="tooltip__message">
+                                    {this.props.tooltipComponent}
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            );
+        }
+        return (
+            <div className="tooltip-wrapper" style={this.props.styles}>
+                <div
+                    ref={(div) => {
+                        this.tooltipContainer = div;
+                    }}>
+                    <div
+                        role="button"
+                        tabIndex="0"
+                        className="tooltip__hover-wrapper"
+                        onBlur={this.closeTooltip}
+                        onFocus={this.showTooltip}
+                        onKeyPress={this.showTooltip}
+                        onMouseEnter={this.showTooltip}
+                        onMouseLeave={this.closeTooltip}
+                        onClick={this.showTooltip}>
+                        {this.props.children}
+                        {this.props.icon && tooltipIcons[this.props.icon]}
+                    </div>
+                    {tooltip}
+                </div>
+            </div>
+        );
+    }
+}
+
+TooltipWrapper.defaultProps = defaultProps;
+TooltipWrapper.propTypes = propTypes;
```

Comment:
Duplicated a ton of code here, hoping we can migrate to this version of the tooltips in the future instead of having a different component for each implementation of a tool tip.

Also thinking it might make this code more readable to have a higher order function that takes two parameters: props for the `TooltipWraper` and any children that would be the hover component for the tool tip.

So like this:

```javascript

export const wrapInToolTip = (tooltipProps, children) => {
  return (
    <TooltipWrapper {...tooltipProps}>
      {children}
    </TooltipWrapper>
  );
}
```

This way when we're nesting all these tooltips it doesn't look so horrible and instead we can just do:

```javascript
wrapInTooltip(tooltipProps, <div style={styles} />)
```